Doc/howto/perf_profiling.rst

```diff
@@ -0,0 +1,201 @@
+.. highlight:: shell-session
+
+.. _perf_profiling:
+
+==============================================
+Python support for the Linux ``perf`` profiler
+==============================================
+
+:author: Pablo Galindo
+
+The Linux ``perf`` profiler is a very powerful tool that allows you to profile and
+obtain information about the performance of your application. It is a very powerful
+profiler that also has a very vibrant ecosystem of tools that aid with the analysis
+of the data that it produces.
+
+The main problem with using the ``perf`` profiler with Python applications is that
+``perf`` only allows to get information about native symbols, this is, the names of
+the functions and procedures written in C. This means that the names and file names
+of the Python functions in your code will not appear in the output of the ``perf``.
+
+Since Python 3.12, the interpreter can run in a special mode that allows Python
+functions to appear in the output of the ``perf`` profiler. When this mode is
+enabled, the interpreter will interpose a small piece of code compiled on the
+fly before the execution of every Python function and it will teach ``perf`` the
+relationship between this piece of code and the associated Python function using
+`perf map files`_.
+
+.. warning::
+
+    Support for the ``perf`` profiler is only currently available for Linux on
+    selected architectures. Check the output of the configure build step or
+    check the output of ``python.exe -m sysconfig | grep HAVE_PERF_TRAMPOLINE``
```

Comment:
Python on Linux does not have an `.exe` suffix.
```suggestion
    check the output of ``python -m sysconfig | grep HAVE_PERF_TRAMPOLINE``
```

Reply:
Unless you build it in docker on an M1. True story